packages/internal/src/files.ts

```diff
@@ -78,51 +78,86 @@ export const findPrerenderedHtml = (cwd = getPaths().web.dist) =>
   fg.sync('**/*.html', { cwd, ignore: ['200.html', '404.html'] })
 
 export const isCellFile = (p: string) => {
-  const { dir, name } = path.parse(p)
-  // A Cell must be a directory named module.
-  if (!dir.endsWith(name)) {
+  // If the path isn't on the web side it cannot be a cell
+  if (!isFileInsideFolder(p, getPaths().web.src)) {
     return false
   }
 
-  const code = fs.readFileSync(p, 'utf-8')
+  // Cell checks parse the file into ast which can fail,
+  // this try-catch block indicates to the user
+  // which file could be the cause of the problem
+  try {
+    const { dir, name } = path.parse(p)
+    // A Cell must be a directory named module.
+    if (!dir.endsWith(name)) {
+      return false
+    }
+
+    const code = fs.readFileSync(p, 'utf-8')
+
+    // A Cell should not have a default export.
+    if (hasDefaultExport(code)) {
+      return false
+    }
+
+    // A Cell must export QUERY and Success.
+    const exports = getNamedExports(code)
+    const exportedQUERY = exports.findIndex((v) => v.name === 'QUERY') !== -1
+    const exportedSuccess =
+      exports.findIndex((v) => v.name === 'Success') !== -1
+    if (!exportedQUERY && !exportedSuccess) {
+      return false
+    }
 
-  // A Cell should not have a default export.
-  if (hasDefaultExport(code)) {
-    return false
+    return true
+  } catch (e) {
+    console.error(`\n Failed to parse file at ${p} \n`)
+    throw e
   }
+}
 
-  // A Cell must export QUERY and Success.
-  const exports = getNamedExports(code)
-  const exportedQUERY = exports.findIndex((v) => v.name === 'QUERY') !== -1
-  const exportedSuccess = exports.findIndex((v) => v.name === 'Success') !== -1
-  if (!exportedQUERY && !exportedSuccess) {
+export const isFileInsideFolder = (filePath: string, folderPath: string) => {
+  const { dir } = path.parse(filePath)
+  const relativePathFromFolder = path.relative(folderPath, dir) //?
+  if (
+    !relativePathFromFolder ||
+    relativePathFromFolder.startsWith('..') ||
+    path.isAbsolute(relativePathFromFolder)
+  ) {
     return false
+  } else {
+    return true
   }
-
-  return true
 }
 
 export const isPageFile = (p: string) => {
-  const { dir, name } = path.parse(p)
+  // Page checks parse the file into ast
+  // Which can fail, this try-catch block indicates to the user
+  // which file could be the cause of the problem
+  try {
+    const { name } = path.parse(p)
+
+    // A page must end with "Page.{jsx,js,tsx}".
+    if (!name.endsWith('Page')) {
+      return false
+    }
+
+    // A page should be in the `web/src/pages` directory.
+    if (!isFileInsideFolder(p, getPaths().web.pages)) {
+      return false
+    }
+
+    // A Page should have a default export.
+    const code = fs.readFileSync(p, 'utf-8')
+    if (!hasDefaultExport(code)) {
+      return false
+    }
```

Comment:
I left a comment on the original issue about why the exception occurs - I believe it valid - is it just that we're not giving a great error message?

So the uncaught exception is originating from this file, right? Maybe we should throw/ console.error inside this function.

Reply:
It'll actually throw in multiple different places here - which is why the try/catch is at this level. The error could happen essentially whenever babel.parse is called

Reply:
Oh and also we don't have access to the path in the functions further down - I think it's cleaner than changing 3/4 functions to take a path just to throw an error

Reply:
Ummm I think in the Cell one... `getNamedExports`